Fix medigun prediction errors

treacherousfiend opened the pull request:
All of the code which determines how much ubercharge you get on a specific tick was server only, causing the medigun to constantly cause prediction errors due to the client having no clue how much uber you were actually building, unless you are in a situation where you are getting the default ubercharge per tick (stock medigun, damaged patient, no critheals, etc.)

This fixes that by making all possible server code also be on the client. The only exception to this is an unused weapon attribute which would modify ubercharge rate as long as you weren't in a respawn room. The functions it calls are server only, and would require changes to how respawn room entities are handled on the client to fix. So any mods, server plugins, vscript, etc. that uses this plugin will still cause prediction errors
Due to being an unused attribute, and because it would require a change to something in a completely different part of the codebase, I decided that it was out of scope for this PR to fix that.

### Related Issue
[Source-1-Games Issue 4046](https://github.com/ValveSoftware/Source-1-Games/issues/4046)

### Implementation
Removed a `#ifdef GAME_DLL` that covered all of the ubercharge rate modification code, the ifdef was left for the `PointInRespawnRoom()` calls due to the function only being on the server.

### Screenshots
<!-- Add screenshots if applicable -->

### Checklist
<!-- You MUST answer "yes" to all of these to open a pull request -->
<!-- To tick a checkbox, place an 'x' in it, like so: [x] -->
- [x] No other PRs implement this idea.
- [x] This PR does not introduce any regressions.
- [x] I certify that this PR is my own entirely original work, or I have permission from and have attributed the relevant authors.
- [x] I have agreed to and signed this project's [Contributor License Agreement](https://cla-assistant.io/mastercomfig/tc2).
- [x] This PR targets the `tc2-mod` branch.

<!-- You do NOT have to answer "yes" to the following, but please mark them if relevant -->
<!-- To tick a checkbox, place an 'x' in it, like so: [x] -->
- [ ] This change has been filed as an issue.
- [x] No other PRs address this.
- [x] This PR is as minimal as possible.
- [x] This PR does not introduce any regressions.
- [x] This PR has been built and locally tested on at least one platform.
- [ ] This PR has been tested on all platforms, on both a dedicated server and on a listen server.

### Testing Checklist
<!-- You do not have to test on all platforms to open a pull request -->
|         |            Client             |            Server             | Version                     |
|---------|:-----------------------------:|:-----------------------------:|-----------------------------|
| Windows | Built & Tested | N/A | Windows 10 Home 22H2 |
|   Linux | Untested | N/A | <!-- `uname -vr` output --> |

### Caveats
The unused attribute "mult_uberchargerate_for_healer" (ubercharge rate bonus for healer) still will cause prediction errors due to relying on server code for func_respawnroom entities

### Alternatives
<!-- Alternatives that were considered -->

treacherousfiend committed:
Fix medigun prediction errors

All of the code which determines how much ubercharge you get on a specific tick was server only, causing the medigun to constantly cause prediction errors unless you are in a situation where you are getting the default ubercharge per tick (stock medigun, damaged patient, no critheals).

This fixes that by making all possible server code to also be on the client.
The only exception to this is an unused weapon attribute which would modify ubercharge rate as long as you weren't in a respawn room. The functions it calls are server only, and would require changes to how respawn room entities are handled on the client to fix.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mastercomfig/tc2?pullRequest=21) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mastercomfig/tc2?pullRequest=21) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mastercomfig/tc2?pullRequest=21) it.</sub>

treacherousfiend committed:
Merge remote-tracking branch 'upstream/master' into Fix-Medigun-Prediction

treacherousfiend committed:
Fix Ubercharge drain prediction errors

Shoutouts to @JustYourOrdinaryNormie for making me aware that my PR didn't fix ubercharge drain prediction errors!

Fixes prediction errors caused by Ubercharge draining by adding client only `DrainCharge`  functioncalls to the medi-gun's `ItemPostFrame` and newly added `ItemBusyFrame` functions.

Additionally, moved the Vaccinator charge removal outside of a server only #ifdef to fix prediction errors from using charges with the Vaccinator.
